### PR TITLE
bugfixes

### PR DESCRIPTION
**Pull request by amirouche**

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
is it possible to add some tests for this change?

**Comment by amirouche** [comment]
Sure!

What tests do you want? It is a frontend matter, so it will require some headless browser thing. Looking at the current repository there seems to be no such thing.

**Comment by amirouche** [comment]
We also need to think how to run the tests inside travis. I have no experience with selenium + travis.

File: sortedm2m/static/sortedm2m/widget.js [hunk 1]
```diff
@@ -19,18 +15,31 @@ if (typeof jQuery === 'undefined') {
                 var label, labelFor;
                 var currentElement = ul;
 
-                while (!label || !label.length) {
+                while (true) {
                     currentElement = currentElement.parent();
+                    if(currentElement[0].tagName == "BODY")
+                        // we can not guess the label.
+                        break;
                     label = currentElement.siblings('label');
```
Comment (amirouche) [inline sortedm2m/static/sortedm2m/widget.js:23]: This will break I think if there is multiple `currentElement.siblings` return several nodes.

Comment (amirouche) [reply sortedm2m/static/sortedm2m/widget.js:23]: Actually no.

**Comment by amirouche** [comment]
> is it possible to add some tests for this change?

@auvipy We are :100: % willing to contribute the necessary code to allow to test frontend aspects. Let me know what you think, in particular what will be your favorite headless browser testing software?

Thanks in advance!

**Review by auvipy — CHANGES_REQUESTED** [review]
you can use qunit?

**Comment by amirouche** [comment]
I added a a minimal example file that demonstrate the bug in the following commit: https://github.com/jazzband/django-sortedm2m/pull/170/commits/08b39757f1a41f7d8f758f18652b4f752c950745

**Comment by amirouche** [comment]
Let me know how do you want to proceed.

**Comment by auvipy** [comment]
pylint --rcfile=pylintrc example sortedm2m sortedm2m_tests test_project *.py

************* Module sortedm2m.forms

sortedm2m/forms.py:23:16: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/forms.py:90:19: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

************* Module sortedm2m.fields

sortedm2m/fields.py:39:27: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:44:21: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:50:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:151:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:187:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:194:35: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:203:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:223:19: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

sortedm2m/fields.py:271:15: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

**Review by auvipy — APPROVED** [review]
Looks good

**Comment by amirouche** [comment]
I tried to silence the only new quality error:

> test_project/qunit-runner.py:4:0: E0611: No name 'webdriver' in module 'selenium' (no-name-in-module)

Seems like the keyword is not `noqa`.

**Comment by clintonb** [comment]
https://github.com/jazzband/django-sortedm2m/pull/172 was merged to fix the pylint issues. Rebase and I'll merge.

**Review by clintonb — APPROVED** [review]

**Pull request merged**